### ISSUE TITLE
Incorrect root node name

### ISSUE THREAD

**Issue by neomonkeus** [niftools/blender_nif_plugin#229]
Plugin is currently not writing the root node name correctly.

```
found bundled python: C:\Program Files\Blender Foundation\Blender\2.74\python
Info: Blender NIF Scripts 2.6.0 (running on Blender 2.74 (sub 0), PyFFI 2.2.2)
Info: exporting nif/object/test_object_export_user_ver.nif
Info: Writing NIF version 0x14000005
Info: Exporting geometry and animation
Info: Smoothing seams between objects...
Info: Fixed normals on 0 vertices.
Info: Exporting
Info: Exporting NiNode block
Info: Exporting objects
Info: Exporting <bpy_struct, Object("NifObject")> as NiNode block
Info: Checking animation groups
Info: Checking controllers
Info: Checking collision
Info: Applying scale correction 1.000000
pyffi.toaster:INFO:--- fix_scale ---
pyffi.toaster:INFO:  scaling by factor 1.000000
pyffi.toaster:INFO:  ~~~ NiNode [unnamed] ~~~
pyffi.toaster:INFO:    ~~~ NiNode [NifObject] ~~~
Info: Writing .nif file
```

**Comment by nightstrike** [comment]
This still appears to be an issue. I'm getting "untitled" as the root name on export, instead of what's in the imported source nif.

**Comment by nightstrike** [comment]
I think a good solution to this would be to put the root nif node in its own empty blender mesh, and put all further nif nodes under that. In Blender's outliner view, then, there'd be a single root node named after the imported nif that contains the entire contents of the nif. This would make it a lot easier for instance to combine multiple meshes, or other such things. But importantly, it would allow retaining (and modifying) the naming information in the blend model.

**Comment by HENDRIX-ZT2** [comment]
The problem is here:
https://github.com/niftools/blender_nif_plugin/blob/67618e3f6da1b87ff7da1b54e7406dbeb265e795/io_scene_nif/nif_export.py#L173
In most cases, the root name should be the name of the armature object or that of the root empty object.
When should it really be the name of the exported file? Are there any actual use cases for that?

**Comment by shaidien** [comment]
It's still broken guys when you going to fix it.
Have you fixed it and not sharing.  Wtf. How hard can it be to not let it move the R L to the end of bone nodes. I guess no one likes skyrim anymore enough to make something work. To bad kind of like blender but without this nif its trash.

**Comment by HENDRIX-ZT2** [comment]
If you explain the problem better and provide example files I should be able to actually fix it. It does not seem related to this particular root node issue. Lots of fixes going on over here: https://github.com/HENDRIX-ZT2/blender_nif_plugin

**Comment by HENDRIX-ZT2** [comment]
Fixed by PR #289. Root node is now always a blender object and thus remembers its name.
